python-threatexchange/threatexchange/exchanges/clients/techagainstterrorism/api.py

```diff
@@ -134,27 +142,52 @@ def get_auth_token(self, username: str, password: str) -> t.Optional[str]:
         )
         return auth_response.get("token")
 
-    def get_hash_list(
-        self, ideology: str = TATIdeology._all.value
-    ) -> t.List[t.Dict[str, str]]:
+    def fetch_hashes(
+        self,
+        order: str = "asc",
+        after: str = "",
+    ) -> TATHashListResponse:
         """
         Get the Hash List JSON file presigned URL ( 5 Minute expiry ) and metadata
         """
 
+        params: t.Dict[str, t.Any] = {
+            "order": order,
+            "after": after,
+        }
+
         try:
             token = self.get_auth_token(self.username, self.password)
-            endpoint = f"{TATEndpoint.hash_list.value}/{ideology}"
+            endpoint = f"{TATEndpoint.hash_list.value}"
 
             logging.info("Fetching TAT hash list")
 
-            # Get the hash list request response
-            response = self._get(endpoint, auth_token=token)
+            results = self._get(endpoint, auth_token=token, **params)
 
-            # Use the pre-signed url from the response to download the hash list values
-            hash_list = self._get(full_url=response["file_url"])
-
-            return hash_list
+            return results
 
         except Exception as exception:
             logging.error("Failed to get hash list: %s", exception)
             raise
+
+    def fetch_hashes_iter(self, next_page: str) -> t.Iterator[TATHashListResponse]:
```

Comment:
It looks like this was copied from the ThreatExchange version of the API, which I think I mostly regretted. It's fine to keep this as is, but it's likely the source of your bug, since it has slightly different behavior resuming vs not. 

blocking: `next_page` looks like it's actually your checkpoint string, aka "after" parameter